src/Parser/Desugar.ml

```diff
@@ -7,33 +7,39 @@
 open Lang.Surface
 
 (** Translation of the binary operator's name to the regular identifier. *)
-let make_bop_id str = "(" ^ str ^ ")"
+let tr_bop_id (op : string node) =
+  if List.mem op.data ["&&"; "||"; ";"] then
+    Error.fatal (Error.desugar_error op.pos)
+  else "(" ^ op.data ^ ")"
```

Comment:
Usage of reserved operators in now prohibited outside of binary expressions.

Reply:
There is a question if `(;) (foo ()) (bar ())` is a valid code, but the decision to allow special operators only is binary expression seems reasonable and I like it. I like the way, how it is handled, i.e., in `tr_bop_id` function which make it works in all places. However, the error messages could be improved (but here, it is not too bad).
```
$ cat test4.dbl
data T = (;)
$ dbl test4.dbl
test4.dbl:1:10-12: fatal error: Syntax error. This construction cannot be used in this context
```